extras/jenkins/Minikube/Jenkinsfile

```diff
@@ -0,0 +1,268 @@
+          expect "Enter Load Balancer End IP:" { send "192.168.100.120\\r" }
+          expect eof
+_EOF_
+            '''
+      }
+    }
+
+    stage('Create VENV') {
+      steps {
+
+        /*
+         * Create our virtual environment.
+         */
+
+          sh '''
+            $WORKSPACE/bin/setup_venv.sh
+            '''
+      }
+    }
+
+    stage('Configure Pulumi') {
+      steps {
+
+        /*
+         * This logic sets the necessary variables in the configuration files; this differs from the manual procedure
+         * where we prompt the user for a number of these required variables. This same approach can be used as part
+         * of the manual deployment if required.
+         *
+         * This will likely evolve further as the project does, and we may reach a point where these defaults are assumed
+         * for a given development type.
+         */
+
+          sh '''
+            echo "PULUMI_STACK=marajenk${BUILD_NUMBER}" > $WORKSPACE/config/pulumi/environment
+            $WORKSPACE/pulumi/python/venv/bin/pulumi stack select --create marajenk${BUILD_NUMBER} -C pulumi/python/config
+            $WORKSPACE/pulumi/python/venv/bin/pulumi stack select --create marajenk${BUILD_NUMBER} -C pulumi/python/kubernetes/applications/sirius
+            $WORKSPACE/pulumi/python/venv/bin/pulumi config set certmgr:helm_timeout "600" -C pulumi/python/config -s marajenk${BUILD_NUMBER}
+            $WORKSPACE/pulumi/python/venv/bin/pulumi config set kic-helm:fqdn "marajenks${BUILD_NUMBER}.zathras.io" -C pulumi/python/config -s marajenk${BUILD_NUMBER}
+            $WORKSPACE/pulumi/python/venv/bin/pulumi config set kic-helm:helm_timeout "600" -C pulumi/python/config -s marajenk${BUILD_NUMBER}
+            $WORKSPACE/pulumi/python/venv/bin/pulumi config set kubernetes:cluster_name "microk8s-cluster" -C pulumi/python/config -s marajenk${BUILD_NUMBER}
+            $WORKSPACE/pulumi/python/venv/bin/pulumi config set kubernetes:infra_type "kubeconfig" -C pulumi/python/config -s marajenk${BUILD_NUMBER}
+            $WORKSPACE/pulumi/python/venv/bin/pulumi config set kubernetes:kubeconfig "$HOME/.kube/config" -C pulumi/python/config -s marajenk${BUILD_NUMBER}
+            $WORKSPACE/pulumi/python/venv/bin/pulumi config set logagent:helm_timeout "600" -C pulumi/python/config -s marajenk${BUILD_NUMBER}
+            $WORKSPACE/pulumi/python/venv/bin/pulumi config set logstore:helm_timeout "600" -C pulumi/python/config -s marajenk${BUILD_NUMBER}
+            $WORKSPACE/pulumi/python/venv/bin/pulumi config set prometheus:adminpass "password" -C pulumi/python/config -s marajenk${BUILD_NUMBER}
+            $WORKSPACE/pulumi/python/venv/bin/pulumi config set prometheus:helm_timeout "600" -C pulumi/python/config -s marajenk${BUILD_NUMBER}
+            $WORKSPACE/pulumi/python/venv/bin/pulumi config set prometheus:helm_timeout "600" -C pulumi/python/config -s marajenk${BUILD_NUMBER}
```

Comment:
Question: for the jenkins build, we assume that 600 will be a good timeout and won't need tweaking?  Is the method we use to adjust it to edit this file and kick off the build again?  Asking mainly out of ignorance since I haven't used Jenkins since ~2013.

Reply:
The good news is that Jenkins hasn't changed much since 2013!!!

Seriously, this is something that needs to be addressed; 600 was basically me throwing a number at the wall - in reality I was seeing it in the 6 minute range, so I bumped it out. This seems to be a good compromise for users who may not have the horsepower of the VM's we're using for testing.